packages/indexer-common/src/indexer-management/monitor.ts

```diff
@@ -610,15 +610,43 @@ export class NetworkMonitor {
         )
       }
 
-      // Check for validity with epoch manager currentEpoch, and fetch block hash
-      const validBlock =
-        result.data.network.latestValidBlockNumber.epochNumber ==
-        (await this.currentEpochNumber())
-          ? result.data.network.latestValidBlockNumber
-          : result.data.network.latestValidBlockNumber.previousBlockNumber
-
-      // Resolve block hash for the given block number.
-      // Calls the configured provider for blocks from protocol chain, or Graph Node otherwise.
+      // Check for validity with Epoch Manager's currentEpoch method, and fetch block hash after that.
+      const epochManagerCurrentEpoch = await this.currentEpochNumber()
+      const epochSubgraphCurrentEpoch =
+        result.data.network.latestValidBlockNumber.epochNumber
+
+      // The Epoch Manager will always lead new epochs, so it should be minuend.
+      const epochDifference = epochManagerCurrentEpoch - epochSubgraphCurrentEpoch
+
+      if (epochDifference < 0) {
+        // Somehow, the Epoch Manager is behind the Epoch Subgraph. This is a critical failure.
+        const criticalErrorMessage = `The Epoch Manager is behind the Epoch Subgraph`
+        this.logger.critical(criticalErrorMessage, {
+          epochManagerCurrentEpoch,
+          epochSubgraphCurrentEpoch,
+        })
+        throw new Error(criticalErrorMessage)
+      }
+
+      if (epochDifference > 1) {
+        // It is not acceptable for the Epoch Subgraph to be more than one epoch away from the
+        // Epoch Manager.
+        const errorMessage = `The Epoch Subgraph is ${epochDifference} epochs away from the Epoch Manager`
```

Comment:
This is a case where we would want indexers to report the issue to us, so we can troubleshoot the epoch block oracle/subgraph. Should we include information on how to report?

Perhaps: 
"The Epoch Subgraph is ${epochDifference} epochs away from the Epoch Manager. It should never be more than 1 epoch away from the Epoch Manager contract. Please submit an issue at https://github.com/graphprotocol/block-oracle/issues/new"

Reply:
Fixed in 1b6b1403